### PR TITLE
Add doc for extended headers

### LINKED ISSUE #2128: Doc update for `-x` flag
Could you please create a doc or update the existing where the usage of `-x` flag will be described?

### PR DESCRIPTION
**Pull request by acid-ant**
Close #2128

Signed-off-by: Anton Nikiforov <an.nikiforov@yadro.com>

### CODE REVIEW

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Existing headers
+
+There are some "well-known" headers starting with `__NEOFS__` prefix that
+affect system behaviour:
+
+* `__NEOFS__NETMAP_EPOCH` - netmap epoch to use for object placement calculation. The `value` is string
+encoded `uint64` in decimal presentation. If set to '0' or not set, the
+current epoch only will be used.
+* `__NEOFS__NETMAP_LOOKUP_DEPTH` - if object can't be found using current epoch's netmap, this header limits
```
Comment (fyrchik) [inline cmd/neofs-cli/docs/storage-node-xheaders.md:19]: I would add that this depth is applied to a current epoch _or the value of `__NEOFS__NETMAP_EPOCH` attribute_ .

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+how many past epochs the node can look up through. The `value` is string
+encoded `uint64` in decimal presentation. If set to '0' or not set, only the
+current epoch will be used.
```
Comment (ikovalev1) [inline cmd/neofs-cli/docs/storage-node-xheaders.md:22]: s/will be/is/

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Responses with duplicated header names or headers with empty values will be
+considered invalid.
+
+## Existing headers
+
+There are some "well-known" headers starting with `__NEOFS__` prefix that
+affect system behaviour:
+
+* `__NEOFS__NETMAP_EPOCH` - netmap epoch to use for object placement calculation. The `value` is string
+encoded `uint64` in decimal presentation. If set to '0' or not set, the
```
Comment (vkarak1) [inline cmd/neofs-cli/docs/storage-node-xheaders.md:17]: Maybe `omitted` instead of `not set`?

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+current epoch only will be used.
+* `__NEOFS__NETMAP_LOOKUP_DEPTH` - if object can't be found using current epoch's netmap, this header limits
+how many past epochs the node can look up through. The `value` is string
+encoded `uint64` in decimal presentation. If set to '0' or not set, only the
+current epoch will be used.
+
+## `neofs-cli` commands with `--xhdr`
+
+List of commands with support of extended headers:
```
Comment (vkarak1) [inline cmd/neofs-cli/docs/storage-node-xheaders.md:26]: I would like to suggest to add at some examples of xheaders usage, but it's up to you.

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Extended headers
+
+## Overview
+
+Extended headers are used for Request/Response. They may contain any user-defined headers
```
[the remark this answers is not in the document]
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md]: Updated.

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+to be interpreted on application level.
+Key name must be a unique valid UTF-8 string. Value can't be empty. Requests or
+Responses with duplicated header names or headers with empty values will be
```
[the remark this answers is not in the document]
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md]: Updated.

File: cmd/neofs-cli/docs/storage-node-xheaders.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+* `__NEOFS__NETMAP_LOOKUP_DEPTH` - if object can't be found using current epoch's netmap, this header limits
+how many past epochs the node can look up through. The `value` is string
+encoded `uint64` in decimal presentation. If set to '0' or not set, only the
+current epoch will be used.
+
+## `neofs-cli` commands with `--xhdr`
+
+List of commands with support of extended headers:
+* `container list-objects`
+* `object delete\get\hash\head\lock\put\range\search`
```
[the remark this answers is not in the document]
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md]: Updated.

File: cmd/neofs-cli/docs/storage-node-xheaders.md (hunk 2 above)
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md]: Updated.

File: cmd/neofs-cli/docs/storage-node-xheaders.md (hunk 3 above)
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md]: Updated.

File: cmd/neofs-cli/docs/storage-node-xheaders.md (hunk 1 above)
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md:19]: Updated.

File: cmd/neofs-cli/docs/storage-node-xheaders.md (hunk 4 above)
Comment (acid-ant) [reply cmd/neofs-cli/docs/storage-node-xheaders.md:26]: Added.

**Review by vkarak1 — APPROVED** [review]

**Review by ikovalev1 — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

**Pull request merged**